client/templates/topic/topicEdit.js

```diff
@@ -44,6 +44,15 @@ function configureSelect2Responsibles() {
     // select the options that where stored with this topic last time
     let topic = getEditTopic();
     if (topic && topic._topicDoc && topic._topicDoc.responsibles) {
+        topic._topicDoc.responsibles.forEach(responsibleId => {
+            let responsibleUser = Meteor.users.findOne(responsibleId);
+            if (!responsibleUser) { //free text user
+                responsibleUser = {fullname: responsibleId};
+            } else {
+                Minutes.formatResponsibles(responsibleUser, 'username', responsibleUser.profile);
+            }
+            selectResponsibles.append('<option value="' + responsibleId + '" selected="selected">' + responsibleUser.fullname + '</option>');
```

Comment:
This makes us open for a possible XSS attack, e.g. if the full name of a responsible user contains HTML/JavaScript. @felixble was already involved in safely handling raw html rendering, he may know more about how to mitigate this.

Reply:
Yes, we had similar issues with our [ConfirmationDialog](https://github.com/4minitz/4minitz/blob/develop/client/helpers/confirmationDialog.js) which could be solved by using the blaze template engine.

Have a look at the function [Blaze.renderWithData(templateOrView, data, parentNode, [nextNode], [parentView])](https://docs.meteor.com/v1.3.5/api/blaze.html#Blaze-renderWithData).

I would suggest to prepare a simple array of responsibles containing the required data (`responsibleId` and `fullname`). Then you could provide a meteor template which consumes such an array and prints for each item an `option`-Element. Then you can use the function `Blaze.renderWithData` by passing the template, the data as well as the element `selectResponsibles` as `parentNode`. You can access the template with its name just like this: `Template['template-name']`.

The meteor docs says "If [the nextNode is] not provided, the template will be inserted as the last child of parentNode." (see [Blaze.renderWithData](https://docs.meteor.com/v1.3.5/api/blaze.html#Blaze-renderWithData) So it should have the same effect as appending strings like you did before. Fortunately, the blaze engine takes care about sanitizing the raw html 👍